processing/src/main/java/org/apache/carbondata/processing/store/TablePage.java

```diff
@@ -65,20 +68,20 @@
     this.pageSize = pageSize;
     keyColumnPage = new KeyColumnPage(pageSize,
         model.getSegmentProperties().getDimensionPartitions().length);
-    noDictDimensionPage = new VarLengthColumnPage[model.getNoDictionaryCount()];
+    noDictDimensionPage = new PrimitiveColumnPage[model.getNoDictionaryCount()];
     for (int i = 0; i < noDictDimensionPage.length; i++) {
-      noDictDimensionPage[i] = new VarLengthColumnPage(pageSize);
+      noDictDimensionPage[i] = new PrimitiveColumnPage(DataType.STRING, pageSize);
```

Comment:
If create another class for string and decimal, then it will be more complex for measure handling. And actually in future I think all simple column handling should be similar, so better to keep them in one class. I think a better class name can solve your comment? How about SimpleColumnPage and ComplexColumnPage?